core/src/main/java/hudson/FilePath.java

```diff
@@ -579,7 +577,7 @@ public Void invoke(File dir, VirtualChannel channel) throws IOException, Interru
         }
         private static final long serialVersionUID = 1L;
     }
-    private static class UnzipLocal extends SecureFileCallable<Void> {
+    private static class UnzipLocal extends MasterToSlaveFileCallable<Void> {
```

Comment:
merits a comment explaining this

Reply:
Per https://github.com/jenkinsci/jenkins/commit/104c751d907919dd53f5090f84d53c671a66457b it actually _is_ transferred: A controller can `untar`/`unzip` a file on an agent to a directory on the agent.

Reply:
Right, just needs a source comment saying that.